src/netgroup.h

```diff
@@ -69,8 +82,17 @@ class NetGroupManager {
      * re-bucketed.
      *
      * This is initialized in the constructor, const, and therefore is
-     * thread-safe. */
-    const std::vector<bool> m_asmap;
+     * thread-safe. m_asmap can either point to m_loaded_asmap which holds
+     * data loaded from an external file at runtime or it can point to embedded
+     * asmap data.
+     */
+    const std::span<const std::byte>  m_asmap;
+    std::vector<std::byte> m_loaded_asmap;
+
+    explicit NetGroupManager(std::span<const std::byte> asmap, std::vector<std::byte> loaded_asmap)
+        : m_asmap(asmap.empty() ? std::span<const std::byte>() : asmap),
+          m_loaded_asmap(std::move(loaded_asmap))
+    {}
```

Comment:
nit: Could also remove stray space and add sanity check in ctor:

```suggestion
    const std::span<const std::byte> m_asmap;
    std::vector<std::byte> m_loaded_asmap;

    explicit NetGroupManager(std::span<const std::byte> asmap, std::vector<std::byte> loaded_asmap)
        : m_asmap(asmap.empty() ? std::span<const std::byte>() : asmap),
          m_loaded_asmap(std::move(loaded_asmap))
    {
        assert(m_loaded_asmap.empty() || m_asmap.data() == m_loaded_asmap.data());
    }
```